### PR TITLE
OE system folders and removing some nodes

### PR DESCRIPTION
**Pull request by llali**
- Remove FileTables and External Tables folders and merge it with Tables
- Add system functions folder to oe and fix the function filters

Fixed issue:

- System folders should only show up under system databases
- Add system database into System Database folder

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/NodeFilter.cs [hunk 1]
```diff
@@ -70,9 +70,14 @@ public string ToPropertyFilterString()
                 {
                     var value = values[i];
                     object proeprtyValue = value;
+                    if (Type == typeof(string))
+                    {
+                        proeprtyValue = $"'{proeprtyValue}'";
```
Comment (kevcunnane) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/NodeFilter.cs]: typo: `propertyValue`

File: src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs [hunk 2]
```diff
@@ -73,22 +82,40 @@ protected virtual void OnExpandPopulateNonFolders(IList<TreeNode> allChildren, T
...
-                        Console.WriteLine("smoObject should not be null");
-                    }
-                    TreeNode childNode = CreateChild(parent, smoObject);
-                    if (childNode != null && PassesFinalFilters(childNode, smoObject) && !ShouldFilterNode(childNode, validForFlag))
+                    foreach (var smoObject in querier.Query(context, propertyFilter, refresh))
                     {
-                        allChildren.Add(childNode);
+                        if (smoObject == null)
+                        {
+                            Console.WriteLine("smoObject should not be null");
```
Comment (kevcunnane) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs]: Should we use Logger here? I think I messed up by using Console?

File: src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/TreeNodeDefinition.xml [hunk 3]
```diff
@@ -308,6 +324,14 @@
       <Filter Property="IsSystemObject" Value="0" Type="bool" />
     </Filters>
   </Node>
+  <Node Name="SystemTableValuedFunctions" LocLabel="SR.SchemaHierarchy_TableValuedFunctions" BaseClass="ModelBased" Strategy="MultipleElementsOfType" ChildQuerierTypes="SqlUserDefinedFunction" TreeNode="TableValuedFunctionTreeNode">
+    <Filters >
+      <Filter Property="FunctionType" Type="Enum" ValidFor="Sql2016|SqlvNext|AzureV12">
```
Comment (kevcunnane) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/TreeNodeDefinition.xml]: All these function types work on SQL 2005+ as far as I know. For both system and non-system folders, I think we need to fix so we no longer filter to just SQL2016 and Azure. Did I mess this up in the original metadata?

File: src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs [hunk 4]
```diff
@@ -73,22 +82,40 @@ protected virtual void OnExpandPopulateNonFolders(IList<TreeNode> allChildren, T
...
+                        if (childNode != null && PassesFinalFilters(childNode, smoObject) && !ShouldFilterNode(childNode, validForFlag))
+                        {
+                            allChildren.Add(childNode);
+                        }
                     }
+
+                }
+                catch (Exception ex)
+                {
+                    string error = string.Format(CultureInfo.InvariantCulture, "Failed getting smo objects. parent:{0} querier: {1} error:{2} inner:{3} stacktrace:{4}",
```
Comment (kevcunnane) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoChildFactoryBase.cs]: Again, long term we need handling of this error handling and somehow return / log in a user-visible way so they can understand failures.

File: src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/TreeNode.cs [hunk 5]
```diff
@@ -300,7 +317,9 @@ protected void PopulateChildren(bool refresh)
             }
             catch(Exception ex)
             {
-                Logger.Write(LogLevel.Error, $"Failed populating oe children. error:{ex.Message} {ex.StackTrace}");
+                string error = string.Format(CultureInfo.InvariantCulture, "Failed populating oe children. error:{0} inner:{1} stacktrace:{2}",
```
Comment (llali) [inline src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/Nodes/TreeNode.cs]: I'll do the error handling in different PR

**Pull request merged**